examples/cat_facts/web-nextjs/src/app/core.ts

```diff
@@ -0,0 +1,114 @@
+import type { Dispatch, SetStateAction } from "react";
+import UAParser from "ua-parser-js";
+
+import { process_event, handle_response, view } from "shared/shared";
+import type {
+  Effect,
+  Event,
+  HttpResponse,
+  KeyValueOutput,
+} from "shared_types/types/shared_types";
+import {
+  PlatformResponse,
+  TimeResponse,
+} from "shared_types/types/shared_types";
+import {
+  EffectVariantRender,
+  ViewModel,
+  EffectVariantHttp,
+  Request,
+  EffectVariantTime,
+  EffectVariantKeyValue,
+  EffectVariantPlatform,
+} from "shared_types/types/shared_types";
+import {
+  BincodeSerializer,
+  BincodeDeserializer,
+} from "shared_types/bincode/mod";
+
+import { request as http } from "./http";
+
+type Response = PlatformResponse | TimeResponse | HttpResponse | KeyValueOutput;
+
+export function update(
+  event: Event,
+  callback: Dispatch<SetStateAction<ViewModel>>
```

Comment:
Does this bind this core wrapper to only work with a `use_state` hook?

Reply:
It _does_. Let me try a callback instead